src/networkWrapper/wrappers/websocket.js

```diff
@@ -1,55 +1,44 @@
-var
-  KuzzleEventEmitter = require('../../eventEmitter');
-
-function WSNode(host, port, ssl) {
-  var self = this;
-  KuzzleEventEmitter.call(this);
-
-  this.WebSocket = typeof WebSocket !== 'undefined' ? WebSocket : require('ws');
-  this.host = host;
-  this.port = port;
-  this.ssl = ssl;
-  this.client = null;
-  this.wasConnected = false;
-  this.retrying = false;
-  this.lasturl = null;
-  this.stopRetryingToConnect = false;
+const
+  RTWrapper = require('./abstract/realtime');
+
+let WebSocketClient;
+
+class WSNode extends RTWrapper {
+
+  constructor(host, options) {
+    super(host, options);
+
+    WebSocketClient = typeof WebSocket !== 'undefined' ? WebSocket : require('ws');
+    this.client = null;
+    this.lasturl = null;
+  }
 
   /**
-   * Creates a new socket from the provided arguments
-   *
-   * @constructor
-   * @param {boolean} autoReconnect
-   * @param {int} reconnectionDelay
-   * @returns {Object} Socket
+   * Connect to the websocket server
    */
-  this.connect = function (autoReconnect, reconnectionDelay) {
-    var
+  connect () {
+    const
       url = (this.ssl ? 'wss://' : 'ws://') + this.host + ':' + this.port,
-      options = typeof window !== 'undefined' ? undefined : {perMessageDeflate: false};
+      opts = typeof window !== 'undefined' ? undefined : {perMessageDeflate: false};
+
+    super.connect();
 
     if (url !== this.lasturl) {
-      self.wasConnected = false;
+      this.wasConnected = false;
       this.lasturl = url;
     }
 
-    this.client = new this.WebSocket(url, options);
+    this.client = new WebSocketClient(url, opts);
 
-    this.client.onopen = function () {
-      if (self.wasConnected) {
-        self.emitEvent('reconnect');
-      }
-      else {
-        self.emitEvent('connect');
-      }
-      self.wasConnected = true;
-      self.stopRetryingToConnect = false;
+    this.client.onopen = () => {
+      this.clientConnected();
     };
 
-    this.client.onclose = function (closeEvent, message) {
-      var error;
-      var status;
-      var reason = message;
+    this.client.onclose = (closeEvent, message) => {
+      let
+        error,
```

Comment:
This `error` variable is shadowed by the one declared in the function starting line 65.

Suggestion: declare this variable as a `const` directly line 58, as it seems to be the only block where it is used.